test/integration-test-main.sh

```diff
@@ -258,7 +261,10 @@ function test_remove_nonempty_directory {
     echo "Testing removing a non-empty directory"
     mk_test_dir
     touch "${TEST_DIR}/file"
-    rmdir "${TEST_DIR}" 2>&1 | grep -q "Directory not empty"
+    if ! (rmdir "${TEST_DIR}" 2>&1 | grep -q "Directory not empty"); then
+        echo "Did not get \"Directory not empty\""
+        exit 1
+    fi
```

Comment:
This change should do the same thing but its not necessary.  I removed it in the last commit (I'll squash commits).  

I was confused because this test is also failing against Amazon S3 servers.  Same problem -- object consistency.  The ```touch testdir/file``` object can show up after the directory has been removed, leaving the "testdir/file" object behind with no directory "testdir" directory object.